src/Integration/MefServices/SharedBindingSuggestionService.cs

```diff
@@ -18,86 +18,90 @@
     internal class SharedBindingSuggestionService : ISharedBindingSuggestionService
     {
-        internal /* for testing */ readonly ConnectConfiguration autobindEnabledConfiguration =
-            new ConnectConfiguration { UseSharedBinding = true };
-
         private readonly ISuggestSharedBindingGoldBar suggestSharedBindingGoldBar;
-        private readonly ITeamExplorerController teamExplorerController;
-        private readonly IConnectedModeWindowEventBasedScheduler connectedModeWindowEventBasedScheduler;
+        private readonly IConnectedModeServices connectedModeServices;
+        private readonly IConnectedModeBindingServices connectedModeBindingServices;
+        private readonly IActiveSolutionTracker activeSolutionTracker;
+        private bool disposed;
 
         [ImportingConstructor]
-        public SharedBindingSuggestionService(ISuggestSharedBindingGoldBar suggestSharedBindingGoldBar,
-            ITeamExplorerController teamExplorerController,
-            IConnectedModeWindowEventBasedScheduler connectedModeWindowEventBasedScheduler)
+        public SharedBindingSuggestionService(
+            ISuggestSharedBindingGoldBar suggestSharedBindingGoldBar,
+            IConnectedModeServices connectedModeServices,
+            IConnectedModeBindingServices connectedModeBindingServices,
+            IActiveSolutionTracker activeSolutionTracker)
         {
             this.suggestSharedBindingGoldBar = suggestSharedBindingGoldBar;
-            this.teamExplorerController = teamExplorerController;
-            this.connectedModeWindowEventBasedScheduler = connectedModeWindowEventBasedScheduler;
+            this.connectedModeServices = connectedModeServices;
+            this.connectedModeBindingServices = connectedModeBindingServices;
+            this.activeSolutionTracker = activeSolutionTracker;
+
+            this.activeSolutionTracker.ActiveSolutionChanged += OnActiveSolutionChanged;
         }
-        
-        public void Suggest(ServerType? serverType, Func<ICommand<ConnectConfiguration>> connectCommandProvider)
+
+        public void Suggest()
         {
-            if (serverType == null)
+            var sharedBindingConfig = connectedModeBindingServices.SharedBindingConfigProvider.GetSharedBinding();
+            var isStandalone = connectedModeServices.ConfigurationProvider.GetConfiguration().Mode == SonarLintMode.Standalone;
+
+            if (sharedBindingConfig?.GetServerType() is { } serverType && isStandalone)
             {
-                return;
+                suggestSharedBindingGoldBar.Show(serverType, ShowManageBindingDialog);
             }
-            
-            suggestSharedBindingGoldBar.Show(serverType.Value, () => ConnectAfterTeamExplorerInitialized(connectCommandProvider));
         }
 
-        public void Close()
+        public void Dispose()
         {
-            suggestSharedBindingGoldBar.Close();
+            Dispose(true);
+            GC.SuppressFinalize(this);
         }
```

Comment:
Sure. But any good reason for this? This is the dispose pattern that is also recommended by Microsoft https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose

Reply:
The `SharedBindingSuggestionService` is not designed and should not be used for inheritance, so it can/should be `sealed`. The above article mentions it https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#cascade-dispose-calls

The other reason is for consistency across our codebase.